File: lib/screens/profile/help_support_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_screenutil/flutter_screenutil.dart';
import '../../theme.dart';

// ════════════════════════════════════════════════════════════════
//  HELP & SUPPORT SCREEN — Figma matched (2095:2847, 390×844)
//
//  Hero: "Center of Operations" fs12 w700 #22D1EE · "HOW CAN WE
//    HELP YOU?" fs48 w700 · Tournament Rules card 342×320
//    #22D1EE@10 (icon 135×143, title fs20 w700, body fs16
//    #FFFFFF@60)
//  Topics: card 342×136 #22D1EE@5 (payments) · 342×102 #22D1EE@5
//    (Account Recovery fs16 w600) · 342×106 #22D1EE@5 (support)
//  FAQ: "Frequently Asked" fs24 w700 + "View All" fs12 #22D1EE ·
//    items 342×var #22D1EE@10
// ════════════════════════════════════════════════════════════════

class HelpSupportScreen extends StatefulWidget {
  const HelpSupportScreen({super.key});

  @override
  State<HelpSupportScreen> createState() => _HelpSupportScreenState();
}

class _HelpSupportScreenState extends State<HelpSupportScreen> {
  int? _expandedFaq;

  final _faqs = const [
    {
      'q': 'How do I enter a tournament?',
      'a':
          'Open the Tournaments tab, pick a tournament, tap PAY & JOIN and confirm your entry fee from your wallet balance.',
    },
    {
      'q': 'When do I get my winnings?',
      'a':
          'Winnings are credited to your wallet immediately after the tournament ends and the leaderboard is verified.',
    },
    {
      'q': 'What are the rules of each game?',
      'a':
          'Fair play and scoring rules are listed in the Tournament Rules section above. Violations may lead to disqualification.',
    },
    {
      'q': 'How do I withdraw my earnings?',
      'a':
          'Go to Wallet → Cash Out, enter the amount, select your bank account and confirm. Transfers process within 24 hours.',
    },
    {
      'q': 'My account was flagged. What do I do?',
      'a':
          'Contact support from this page with your username. Our team reviews flagged accounts within 48 hours.',
    },
  ];

  @override
  Widget build(BuildContext context) => Scaffold(
    backgroundColor: context.bg,
    body: SafeArea(
      child: Column(children: [
        // Header — Figma Frame 56
        Container(
          width: double.infinity,
          padding: EdgeInsets.fromLTRB(24.w, 40.h, 24.w, 16.h),
          decoration: BoxDecoration(
            color: context.bg,
            border: Border(bottom: BorderSide(color: context.border, width: 1)),
          ),
          child: Row(children: [
            GestureDetector(
              onTap: () => Navigator.maybePop(context),
              child: Icon(Icons.close_rounded,
                  color: context.txtPri, size: 20.w),
            ),
            Expanded(
              child: Text('Help & Support',
                  textAlign: TextAlign.center,
                  style: TextStyle(
                      color: context.txtPri,
                      fontSize: 18.sp, fontWeight: FontWeight.w700)),
            ),
            SizedBox(width: 20.w),
          ]),
        ),

        Expanded(
          child: ListView(
            padding: EdgeInsets.symmetric(horizontal: 24.w),
            children: [

              SizedBox(height: 32.h),

              // ── HERO ──────────────────────────────────────────────────
              Text('Center of Operations',
                  style: TextStyle(
                      color: kCyan, fontSize: 12.sp,
                      fontWeight: FontWeight.w700)),
              SizedBox(height: 8.h),
              Text('HOW CAN WE\nHELP YOU?',
                  style: TextStyle(
                      color: context.txtPri,
                      fontSize: 38.sp, fontWeight: FontWeight.w700,
                      height: 1.1)),
              SizedBox(height: 20.h),

              // Tournament Rules card — Figma: 342×320 rx12 pad[32,32,32,32]
              Container(
                padding: EdgeInsets.all(32.r),
                decoration: BoxDecoration(
                  color: kCyan.withOpacity(0.1),
                  borderRadius: BorderRadius.circular(12.r),
                ),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Container(
                      width: 72.w, height: 72.h,
                      decoration: BoxDecoration(
                        color: kCyan.withOpacity(0.15),
                        borderRadius: BorderRadius.circular(20.r),
                      ),
                      child: Icon(Icons.emoji_events_outlined,
                          color: kCyan, size: 40.w),
                    ),
                    SizedBox(height: 18.h),
                    Text('Tournament Rules',
                        style: TextStyle(
                            color: context.txtPri,
                            fontSize: 20.sp, fontWeight: FontWeight.w700)),
                    SizedBox(height: 8.h),
                    Text(
                      'Master the arena. Everything you\nneed to know about fair play and\nscoring.',
                      style: TextStyle(
                          color: context.txtSec, fontSize: 16.sp,
                          fontWeight: FontWeight.w400, height: 1.45),
                    ),
                  ],
                ),
              ),

              SizedBox(height: 24.h),

              // ── TOPIC CARDS ───────────────────────────────────────────
              _TopicCard(
                icon: Icons.account_balance_wallet_outlined,
                title: 'Payments & Withdrawals',
                sub: 'Secure withdrawals and credit\nprocessing.',
              ),
              SizedBox(height: 12.h),
              _TopicCard(
                icon: Icons.restore_outlined,
                title: 'Account Recovery',
                sub: 'Reset your password or recover a\nlocked account.',
                compact: true,
              ),
              SizedBox(height: 12.h),
              _TopicCard(
                icon: Icons.support_agent_outlined,
                title: 'Game Support',
                sub: 'Report a bug or talk to our\nsupport team.',
                compact: true,
              ),

              SizedBox(height: 24.h),

              // ── FAQ — Figma: "Frequently Asked" fs24 w700 CENTER ────
              Text('Frequently Asked',
                  textAlign: TextAlign.center,
                  style: TextStyle(
                      color: context.txtPri,
                      fontSize: 24.sp, fontWeight: FontWeight.w700)),
              SizedBox(height: 14.h),

              ...List.generate(_faqs.length, (i) {
                final expanded = _expandedFaq == i;
                return Container(
                  margin: EdgeInsets.only(bottom: 12.h),
                  decoration: BoxDecoration(
                    color: kCyan.withOpacity(0.1),
                    borderRadius: BorderRadius.circular(12.r),
                  ),
                  child: Theme(
                    data: Theme.of(context)
                        .copyWith(dividerColor: Colors.transparent),
                    child: ExpansionTile(
                      tilePadding: EdgeInsets.symmetric(
                          horizontal: 16.w, vertical: 2.h),
                      childrenPadding:
                          EdgeInsets.fromLTRB(16.w, 0.h, 16.w, 16.h),
                      title: Text(_faqs[i]['q']!,
                          style: TextStyle(
                              color: context.txtPri,
                              fontSize: 14.sp,
                              fontWeight: FontWeight.w600)),
                      trailing: Icon(
                        expanded
                            ? Icons.keyboard_arrow_up_rounded
                            : Icons.keyboard_arrow_down_rounded,
                        color: context.txtSec,
                      ),
                      onExpansionChanged: (open) {
                        setState(() =>
                            _expandedFaq = open ? i : null);
                      },
                      children: [
                        Align(
                          alignment: Alignment.centerLeft,
                          child: Text(_faqs[i]['a']!,
                              style: TextStyle(
                                  color: context.txtSec,
                                  fontSize: 13.sp, height: 1.5)),
                        ),
                      ],
                    ),
                  ),
                );
              }),

              SizedBox(height: 32.h),
            ],
          ),
        ),
      ]),
    ),
  );
}

// ── TOPIC CARD — Figma: 342×136/102 #22D1EE@5 ────────────────────
class _TopicCard extends StatelessWidget {
  final IconData icon;
  final String title;
  final String sub;
  final bool compact;

  const _TopicCard({
    required this.icon,
    required this.title,
    required this.sub,
    this.compact = false,
  });

  @override
  Widget build(BuildContext context) => Container(
    padding: EdgeInsets.symmetric(
        horizontal: (compact ? 16 : 16).w, vertical: (compact ? 24 : 16).h),
    decoration: BoxDecoration(
      color: kCyan.withOpacity(0.05),
      borderRadius: BorderRadius.circular(8.r),
    ),
    child: Row(children: [
      Container(
        width: 48.w, height: 48.h,
        decoration: BoxDecoration(
          color: kCyan.withOpacity(0.12),
          borderRadius: BorderRadius.circular(12.r),
        ),
        child: Icon(icon, color: kCyan, size: 24.w),
      ),
      SizedBox(width: 16.w),
      Expanded(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(title,
                style: TextStyle(
                    color: context.txtPri,
                    fontSize: 16.sp, fontWeight: FontWeight.w600)),
            SizedBox(height: 4.h),
            Text(sub,
                style: TextStyle(
                    color: context.txtSec, fontSize: 12.sp,
                    fontWeight: FontWeight.w500, height: 1.4)),
          ],
        ),
      ),
      Icon(Icons.chevron_right_rounded, color: context.txtSec),
    ]),
  );
}

File: src/screens/profile/HelpSupportScreen.js
import React, { useState, useEffect } from 'react';
import {
  View,
  Text,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  StatusBar,
  ActivityIndicator,
} from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import {
  ArrowLeft,
  Swords,
  Wallet,
  Key,
  LifeBuoy,
  ChevronDown,
} from 'lucide-react-native';
import { useTheme } from '../../context/ThemeContext';
import { content } from '../../services/api';

const FALLBACK_SECTIONS = [
  {
    id: 'games',
    title: 'Tournament Rules',
    items: [
      {
        question: 'Can I play for free?',
        answer:
          'Yes. Every game has a practice mode against the Gamearn bot with no entry fee. Real-money games require a matchmaking entry fee from your wallet.',
      },
      {
        question: 'How does the streak work?',
        answer:
          'Play (or win) at least one game each day to grow your streak. Streaks are saved to your profile so they persist across devices. Miss a day and the streak resets, unless you recover it with the ad option.',
      },
    ],
  },
  {
    id: 'wallet',
    title: 'Wallet & Payments',
    items: [
      {
        question: 'How do I withdraw my balance?',
        answer:
          'Open Wallet > Withdraw, enter your Nigerian bank details and amount. For security you may be asked to re-authenticate. Withdrawals are processed and usually arrive within 24 hours.',
      },
      {
        question: 'What is the daily free bonus?',
        answer:
          'A small coin bonus you can claim once every 24 hours from the bonus button. Coins are practice currency used for friendly games — they are not real money and cannot be withdrawn.',
      },
    ],
  },
  {
    id: 'account',
    title: 'Account Recovery',
    items: [
      {
        question: 'I forgot my password.',
        answer:
          'On the login screen tap "Forgot Password" and enter your email. We will send you a reset link straight from Firebase, which lets you choose a new password.',
      },
      {
        question: 'How do I enable 2FA?',
        answer:
          'Go to Settings > Account Security. Choose email or phone as your factor, verify the code we send, and toggle 2FA on. Email 2FA works in the current build; phone 2FA uses SMS and is available in the full native build.',
      },
    ],
  },
];

const SECTION_ICONS = {
  games: Swords,
  wallet: Wallet,
  account: Key,
  support: LifeBuoy,
};

function titleCase(str) {
  if (!str) return '';
  return str
    .split(' ')
    .map((w) => (w ? w[0].toUpperCase() + w.slice(1) : w))
    .join(' ');
}

export default function HelpSupportScreen({ navigation }) {
  const { theme, isDark } = useTheme();
  const [sections, setSections] = useState(FALLBACK_SECTIONS);
  const [expanded, setExpanded] = useState({});
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    let mounted = true;
    (async () => {
      try {
        const list = await content.help();
        if (mounted && Array.isArray(list) && list.length > 0) {
          setSections(list);
        }
      } catch {
        // Offline or server unreachable — fall back to local content.
      } finally {
        if (mounted) setLoading(false);
      }
    })();
    return () => {
      mounted = false;
    };
  }, []);

  const toggleItem = (secIdx, itemIdx) => {
    const key = `${secIdx}-${itemIdx}`;
    setExpanded((prev) => ({ ...prev, [key]: !prev[key] }));
  };

  const items = sections.map((section, secIdx) => ({
    ...section,
    items: (section.items || []).map((item, itemIdx) => ({
      ...item,
      key: `${secIdx}-${itemIdx}`,
    })),
  }));

  return (
    <View style={[styles.screenRoot, { backgroundColor: theme.bg }]}>
      <StatusBar barStyle={theme.statusBar} backgroundColor={theme.bg} />
      <LinearGradient colors={theme.gradientBg} style={StyleSheet.absoluteFillObject} />

      {/* Header */}
      <View style={styles.topHeader}>
        <TouchableOpacity
          onPress={() => (navigation.canGoBack() ? navigation.goBack() : navigation.navigate('MainTabs'))}
          style={[styles.backCircleBtn, { backgroundColor: isDark ? 'rgba(255, 255, 255, 0.08)' : 'rgba(0, 0, 0, 0.05)' }]}
        >
          <ArrowLeft size={20} color={theme.textPrimary} />
        </TouchableOpacity>
        <Text style={[styles.headerTitle, { color: theme.textPrimary }]}>Help & Support</Text>
        <View style={{ width: 40 }} />
      </View>

      <ScrollView contentContainerStyle={styles.scrollContent} showsVerticalScrollIndicator={false}>
        {/* Hero Title Section */}
        <View style={styles.heroTitleBlock}>
          <Text style={styles.centerOpsLabel}>CENTER OF OPERATIONS</Text>
          <Text style={[styles.mainHeroTitle, { color: theme.textPrimary }]}>HOW CAN WE</Text>
          <Text style={styles.orangeHeroTitle}>HELP YOU?</Text>
        </View>

        {loading ? (
          <View style={styles.loadingBox}>
            <ActivityIndicator color={theme.primary} />
            <Text style={[styles.loadingText, { color: theme.textMuted }]}>Loading help articles...</Text>
          </View>
        ) : (
          items.map((section, secIdx) => {
            const Icon = SECTION_ICONS[section.id] || LifeBuoy;
            const isFirst = secIdx === 0;
            const sectionCardStyle = isFirst ? styles.categoryCardHero : styles.categoryCardRow;
            const sectionTitle = titleCase(section.title);
            return (
              <View
                key={section.id || secIdx}
                style={[sectionCardStyle, { backgroundColor: theme.cardBg, borderColor: isFirst ? theme.cardBorder : theme.cardBorderSubtle }]}
              >
                <View style={styles.sectionHeaderRow}>
                  <View style={[styles.iconCircleCyan, !isFirst && styles.iconCircleCyanSmall]}>
                    <Icon size={isFirst ? 24 : 20} color="#00E5FF" />
                  </View>
                  <View style={styles.rowLeftContent}>
                    <Text style={[styles.cardTitle, isFirst && styles.cardTitleSmall, { color: theme.textPrimary }]}>
                      {sectionTitle.toUpperCase()}
                    </Text>
                  </View>
                </View>

                {section.items.slice(0, 5).map((item) => {
                  const open = !!expanded[item.key];
                  return (
                    <View key={item.key} style={styles.faqItem}>
                      <TouchableOpacity
                        activeOpacity={0.8}
                        onPress={() => toggleItem(secIdx, parseInt(item.key.split('-')[1], 10))}
                        style={styles.faqQuestionRow}
                      >
                        <Text style={[styles.faqQuestion, { color: theme.textPrimary }]}>
                          {item.question || item.title || 'Topic'}
                        </Text>
                        <ChevronDown size={18} color={theme.textMuted} style={[styles.faqChevron, open && styles.faqChevronOpen]} />
                      </TouchableOpacity>
                      {open && (
                        <Text style={[styles.faqAnswer, { color: theme.textSecondary }]}>
                          {item.answer || item.body || 'No details available yet.'}
                        </Text>
                      )}
                    </View>
                  );
                })}
              </View>
            );
          })
        )}
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  screenRoot: {
    flex: 1,
  },
  topHeader: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 20,
    paddingTop: 60,
    paddingBottom: 15,
  },
  backCircleBtn: {
    width: 40,
    height: 40,
    borderRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
  headerTitle: {
    fontSize: 22,
    fontWeight: '800',
  },
  scrollContent: {
    paddingHorizontal: 20,
    paddingBottom: 40,
  },
  heroTitleBlock: {
    marginTop: 10,
    marginBottom: 28,
  },
  centerOpsLabel: {
    color: '#00E5FF',
    fontSize: 11,
    fontWeight: '900',
    letterSpacing: 1,
    marginBottom: 6,
  },
  mainHeroTitle: {
    fontSize: 34,
    fontWeight: '900',
    lineHeight: 38,
  },
  orangeHeroTitle: {
    color: '#FF5500',
    fontSize: 36,
    fontWeight: '900',
    lineHeight: 40,
  },
  categoryCardHero: {
    borderRadius: 24,
    borderWidth: 1,
    padding: 24,
    marginBottom: 16,
    position: 'relative',
    overflow: 'hidden',
  },
  iconCircleCyan: {
    width: 52,
    height: 52,
    borderRadius: 18,
    backgroundColor: 'rgba(0, 229, 255, 0.1)',
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 16,
  },
  cardTitle: {
    fontSize: 20,
    fontWeight: '900',
    letterSpacing: 0.5,
    marginBottom: 8,
  },
  cardSubText: {
    fontSize: 14,
    lineHeight: 20,
    maxWidth: '85%',
  },
  decorLinesPattern: {
    position: 'absolute',
    right: -10,
    bottom: -10,
    gap: 6,
    transform: [{ rotate: '-45deg' }],
  },
  decorLineBar: {
    width: 50,
    height: 6,
    borderRadius: 3,
    backgroundColor: 'rgba(255, 255, 255, 0.04)',
  },
  categoryCardRow: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    borderRadius: 20,
    borderWidth: 1,
    padding: 18,
    marginBottom: 14,
  },
  rowLeftContent: {
    flex: 1,
    paddingRight: 10,
  },
  rowHeaderInline: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 4,
  },
  rowTitleText: {
    fontSize: 15,
    fontWeight: '900',
    letterSpacing: 0.5,
  },
  rowSubText: {
    fontSize: 12,
    lineHeight: 16,
  },
  loadingBox: {
    alignItems: 'center',
    paddingVertical: 40,
    gap: 10,
  },
  loadingText: {
    fontSize: 13,
  },
  sectionHeaderRow: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 6,
  },
  iconCircleCyanSmall: {
    width: 40,
    height: 40,
    borderRadius: 13,
    marginBottom: 0,
    marginRight: 12,
  },
  cardTitleSmall: {
    fontSize: 16,
    marginBottom: 0,
  },
  faqItem: {
    marginTop: 4,
    borderTopWidth: StyleSheet.hairlineWidth,
    borderTopColor: 'rgba(128, 128, 128, 0.2)',
    paddingVertical: 2,
  },
  faqQuestionRow: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingVertical: 12,
    gap: 8,
  },
  faqQuestion: {
    flex: 1,
    fontSize: 13,
    fontWeight: '700',
  },
  faqChevron: {
    transform: [{ rotate: '0deg' }],
  },
  faqChevronOpen: {
    transform: [{ rotate: '180deg' }],
  },
  faqAnswer: {
    fontSize: 13,
    lineHeight: 19,
    paddingBottom: 12,
  },
});
